apiserver/common/modelstatus.go

```diff
@@ -93,28 +93,29 @@ func (c *ModelStatusAPI) modelStatus(tag string) (params.ModelStatus, error) {
 		return status, errors.Trace(err)
 	}
 
-	volumes, err := st.AllVolumes()
-	if err != nil {
-		return status, errors.Trace(err)
-	}
-	modelVolumes := ModelVolumeInfo(volumes)
-
-	filesystems, err := st.AllFilesystems()
-	if err != nil {
-		return status, errors.Trace(err)
-	}
-	modelFilesystems := ModelFilesystemInfo(filesystems)
-
-	return params.ModelStatus{
+	result := params.ModelStatus{
 		ModelTag:           tag,
 		OwnerTag:           model.Owner().String(),
 		Life:               params.Life(model.Life().String()),
 		HostedMachineCount: len(hostedMachines),
 		ApplicationCount:   len(applications),
 		Machines:           modelMachines,
-		Volumes:            modelVolumes,
-		Filesystems:        modelFilesystems,
-	}, nil
+	}
+
+	if model.Type() == state.ModelTypeIAAS {
```

Comment:
100% sure that this will work all the time give recent difficulties to figure out what to do on older versions?
(I guess, by the time, this is called upgrades steps would have ensured valid data here?..)

Reply:
Right, this should be covered by upgrades. We don't protect this call anywhere.
The recent issue was due to the IAASModel() call which does return an error.